### PR TITLE
State Management Intro - Leigha updates

### PR DESCRIPTION
**Pull request by PoojaB26**
(cherry picked from commit 5611b5ba68a41ad03110c0495dcb9501df64038e)

### Description
Provide a brief overview of what this documentation update is about. Explain what sections or topics are being added or revised.

Linear ticket and [magic word](https://linear.app/docs/github#link-prs) Fixes DEVR-337  

## Type of change
- [ ] Typo fix 
- [ ] New feature 
- [ ] Removed outdated references

### CODE REVIEW

File: docs/concepts/state-management.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+![App State is shared across multiple pages in the application. Component State is specific to a component. Page State is shared across widgets on the page.](<state_management_overview.png>)
+
+
+- State variables are themselves variables - meaning they have a *name* and a *data type*. 
+- They also have an initial value that is set when you create the variable. 
+- Once you create a state variable, it's value can be used to change the configuration of widget properties - like any other variable. 
+- You can update the value of state variables using the *Update State Variable* action.
+
+You can learn more about state management from this video:
+<div class="video-container"><iframe src="https://www.youtube.com/watch?v=jD6L4xjYjJA" title="YouTube video player"  frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe></div>
```
Comment (PoojaB26) [inline docs/concepts/state-management.md:22]: Video seems to be broken on running the site, can you please check?

File: docs/concepts/state-management.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+---
+
+State management is a crucial concept focused on maintaining and controlling the *state* of an application. Simply put, it involves monitoring the changes within your app and updating the user interface to reflect these changes. 
+
+You can think of the UI as a function of your state variables, where changes in these variables directly influence what the UI displays.
+
+In FlutterFlow, there are a few types of state variables that you can create:
+
+
+![App State is shared across multiple pages in the application. Component State is specific to a component. Page State is shared across widgets on the page.](<state_management_overview.png>)
```
Comment (PoojaB26) [inline docs/concepts/state-management.md:13]: The alt text does not show up on the UI actually. Can we add a captioned image with that text as well? 

Please [check point 16 in Writing Guidelines doc](https://www.notion.so/flutterflow/Docs-Contribution-Guidelines-42829c305df441f6b061b4095abac8a8) for how to add one!

File: sidebars.ts [hunk 3]
```diff
@@ -2,42 +2,23 @@ import type {SidebarsConfig} from '@docusaurus/plugin-content-docs';
...
+      items: ['index', {type: 'category', label: "Before You Begin", items: [
+        'intro/before-you-begin/setting-up-flutterflow', 'intro/before-you-begin/app-architecture'
+    ]}]},
+    {
+      type: 'category',
+      label: 'Concepts',
+      collapsed: false,
+      items: ['concepts/state-management']
+    }
+  ]
```
Comment (PoojaB26) [inline sidebars.ts:21]: Can you please review this change? It removes rest of the folders form UI such as Resources, Building Concepts (that already has a sub folder called State Management that you can move your files to).

We can call them Building Concepts or just Concepts, I don't have any preference!

Also for any non-intro folder, you can simply do autogenerated folders so we don't have to explicitly mention the sub folder names. 

Example: 
```
items: [
                {
                    type: 'autogenerated',
                    dirName: 'ff-concepts'
                }
            ],
 ```

Comment (PoojaB26) [reply sidebars.ts:21]: So if we want to change from `ff-concepts` to just `concepts`, then lets just change the already existing folder name and change the dirName above (also already present in sidebar file)

**Review by PoojaB26 — COMMENTED** [review]
@leighajarett Added some comments here

File: docs/concepts/state-management.md (hunk 1 above)
Comment (leighajarett) [reply docs/concepts/state-management.md]: Hmm I tried copying the URL from embed in YouTube but can't get it to work, any ideas?

https://www.youtube.com/watch?v=jD6L4xjYjJA

File: docs/concepts/state-management.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+description: An overview of state management in FlutterFlow
+---
+
+State management is a crucial concept focused on maintaining and controlling the *state* of an application. Simply put, it involves monitoring the changes within your app and updating the user interface to reflect these changes. 
+
+You can think of the UI as a function of your state variables, where changes in these variables directly influence what the UI displays.
+
+In FlutterFlow, there are a few types of state variables that you can create:
+
+<figure>
```
Comment (pinkeshmars) [inline docs/concepts/state-management.md:12]: Maybe we increase the text size in the image? For me, it looks smaller than the doc text size and is hard to read.

File: docs/concepts/state-management.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+
+- State variables are themselves variables - meaning they have a *name* and a *data type*. 
+- They also have an initial value that is set when you create the variable. 
+- Once you create a state variable, it's value can be used to change the configuration of widget properties - like any other variable. 
+- You can update the value of state variables using the *Update State Variable* action.
+
+You can learn more about state management from this video:
+<div class="video-container">
+<iframe src="https://www.youtube.com/embed/jD6L4xjYjJA?si=-RjniUB-K0ZsMoB1" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
```
Comment (pinkeshmars) [inline docs/concepts/state-management.md:25]: The video looks very small. Can you try adding an iframe size like '<iframe width="760" height="428" '?

**Review by pinkeshmars — CHANGES_REQUESTED** [review]

**Pull request merged**